src/test/java/bio/terra/tanagra/query/FieldVariableTest.java

```diff
@@ -0,0 +1,96 @@
+package bio.terra.tanagra.query;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.category.Unit;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag(Unit.TAG)
+class FieldVariableTest {
+
+  @Test
+  void renderSQL() {
+    var fieldPointer = new FieldPointer.Builder().columnName("field").build();
+    var tableVariable = TableVariable.forPrimary(TablePointer.fromTableName(null, "table"));
+    TableVariable.generateAliases(List.of(tableVariable));
+    assertThat(new FieldVariable(fieldPointer, tableVariable).renderSQL(null), is("t.field"));
+
+    assertThat(
+        new FieldVariable(fieldPointer, tableVariable, "bar").renderSQL(null),
+        is("t.field AS bar"));
+
+    var fieldPointerForeignKey =
+        new FieldPointer.Builder().foreignTablePointer(TablePointer.fromRawSql(null, null)).build();
+    var fieldVariableForeignKey = new FieldVariable(fieldPointerForeignKey, tableVariable);
+    assertThrows(
+        UnsupportedOperationException.class,
+        () -> fieldVariableForeignKey.renderSQL(SqlPlatform.BIGQUERY));
+
+    var fieldVariableFunctionWrapper =
+        new FieldVariable(
+            new FieldPointer.Builder().sqlFunctionWrapper("foo").columnName("field").build(),
+            tableVariable,
+            "alias");
+    assertThat(fieldVariableFunctionWrapper.renderSQL(null), is("foo(t.field)"));
+
+    var fieldVariableSqlFunctionWrapper =
+        new FieldVariable(
+            new FieldPointer.Builder()
+                .sqlFunctionWrapper("custom(<fieldSql>)")
+                .columnName("field")
+                .build(),
+            tableVariable,
+            "alias");
+    assertThat(fieldVariableSqlFunctionWrapper.renderSQL(null), is("custom(t.field)"));
```

Comment:
Do you think it would be worth breaking this test up and creating individual tests for each assertion?

Reply:
Sure, I was planning to revisit this once I had more tests written; the way the code sets up `FieldPointer` and `TablePointer` is pretty verbose right now. In the original code these objects are only constructed using serialization so no much effort was given to manual construction.